src/arti/tor-dirmgr/src/authority.rs

```diff
@@ -44,6 +46,25 @@ impl Authority {
     }
 }
 
+impl FromStr for Authority {
+    type Err = Error;
+
+    /// Parse Authority from a string.
+    fn from_str(authority_raw: &str) -> Result<Self, Self::Err> {
+        // name, v3ident_raw
+        let authority: Vec<&str> = authority_raw.split_whitespace().collect();
+
+        let name = authority[0];
+        let v3ident_raw = authority[1];
```

Comment:
you don't check that it indeed has the correct number of elements, so it will panic if it is badly formatted

Reply:
Thank you for the tip with `bail!(...)`, I didn't know this macro.